### ISSUE TITLE
fix examples in modeling_procedure

### ISSUE THREAD

**Issue by jangorecki** [Molmed/emil#4]
There is an attempt to load `caret` namespace in examples [modeling_procedure.r#L27](https://github.com/Molmed/emil/blob/master/examples/modeling_procedure.r#L27), code below is the results from examples.
As `caret` is suggested dependency, it should not be required to run examples according to [R-exts](https://cran.r-project.org/doc/manuals/R-exts.html#Suggested-packages). This can be conditionally escaped with `requireNamespace`.
It is possible there are other cases of that issue in other examples, this one was to first that fails.

```
* checking examples ... ERROR
Running examples in ‘emil-Ex.R’ failed
The error most likely occurred in:

> base::assign(".ptime", proc.time(), pos = "CheckExEnv")
> ### Name: modeling_procedure
> ### Title: Setup a modeling procedure
> ### Aliases: modeling_procedure
> 
> ### ** Examples
> 
> # 1: Fit linear discriminants without tuning any parameter,
> # since it has none
> modeling_procedure("lda")
`lda` modeling procedure.

   model fitting function:       yes
   prediction function:          yes
   feature importance function:  no
   individual error function:    no

   number of parameter sets to tune over: 1
   tuned: not needed
> 
> # 2: Tune random forest's `mtry` parameter, with 3 possible values
> modeling_procedure("randomForest", list(mtry = list(100, 250, 1000)))
`randomForest` modeling procedure.

   model fitting function:       yes
   prediction function:          yes
   feature importance function:  yes
   individual error function:    no

   number of parameter sets to tune over: 3
   tuned: no
> 
> # 3: Tune random forest's `mtry` and `maxnodes` parameters simultaneously,
> # with 3 values each, testing all 9 possible combinations
> modeling_procedure("randomForest", list(mtry = list(100, 250, 1000),
+                                         maxnodes = list(5, 10, 25)))
`randomForest` modeling procedure.

   model fitting function:       yes
   prediction function:          yes
   feature importance function:  yes
   individual error function:    no

   number of parameter sets to tune over: 9
   tuned: no
> 
> # 4: Tune random forest's `mtry` and `maxnodes` parameters simultaneously,
> # but only test 3 manually specified combinations of the two
> modeling_procedure("randomForest", list(list(mtry = 100, maxnodes = 5),
+                                    list(mtry = 250, maxnodes = 10),
+                                    list(mtry = 1000, maxnodes = 25)))
`randomForest` modeling procedure.

   model fitting function:       yes
   prediction function:          yes
   feature importance function:  yes
   individual error function:    no

   number of parameter sets to tune over: 3
   tuned: no
> 
> # 5: Tune elastic net's `alpha` and `lambda` parameters. Since elastic net's
> # fitting function can tune `lambda` internally in a more efficient way
> # than the general framework is able to do, only tune `alpha` and pass all
> # `lambda` values as a single argument.
> modeling_procedure("glmnet", list(alpha = seq(0, 1, length.out=6),
+                                   lambda = list(seq(0, 5, length.out=30))))
`glmnet` modeling procedure.

   model fitting function:       yes
   prediction function:          yes
   feature importance function:  yes
   individual error function:    no

   number of parameter sets to tune over: 6
   tuned: no
> 
> # 6: Train elastic nets using the caret package's model fitting framework
> library(caret)
Error in library(caret) : there is no package called ‘caret’
Execution halted
```

**Comment by jangorecki** [comment]
similarly `cmprsk` in `pvalue.crr`

```
* checking examples ... ERROR
Running examples in ‘emil-Ex.R’ failed
The error most likely occurred in:

> base::assign(".ptime", proc.time(), pos = "CheckExEnv")
> ### Name: pvalue.crr
> ### Title: Extracts p-value from a competing risk model
> ### Aliases: pvalue.crr
> 
> ### ** Examples
> 
> library(cmprsk)
Error in library(cmprsk) : there is no package called ‘cmprsk’
Execution halted
```

**Comment by backlin** [comment]
I've corrected all imports now (ce496a620807e6562259eeed8c73ce75ad68714f). 

Thanks for checking the package!
